Engine/DataFeeds/Enumerators/DelistingEventProvider.cs

```diff
@@ -58,6 +59,12 @@ public class DelistingEventProvider : ITradableDateEventProvider
                     _delistingDate = config.Symbol.ID.Date;
                     break;
                 case SecurityType.Option:
+                    if (config.Symbol.Underlying.SecurityType == SecurityType.Future)
+                    {
+                        _delistingDate = FutureOptionSymbol.GetLastDayOfTrading(config.Symbol);
```

Comment:
There is another case of this at the `SubscriptionDataReader` to be fixed, suggest we add some extension method to handle this logic?